tests/unit/schema/test.commands.js

```diff
@@ -32,7 +32,7 @@ describe('/commands', () => {
     validateAddon(manifest);
     assertHasMatchingError(validateAddon.errors, {
       dataPath: '/commands/up/suggested_key/mac',
-      message: /should match pattern/,
+      message: 'should match format "manifestShortcutKey"',
```

Comment:
uhm... I'm starting to think that the change to the manifest's shortcut keys used in the manifest is going to produce validation errors that may not be always immediately "actionable", the previous version was emitting a "monstrous" validation error that wasn't super-readable but it still listed the allowed patterns in a way that the addon developer could (more or less :-)) use to understand which are the allowed keys.

We may need to think how we can produce some better error messages now that the validation is being done from the js code instead of patterns included in the schemas.

Reply:
This is really hard tbh and there's also https://github.com/mozilla/addons-linter/issues/1905 which links to the same issue.

Personally, I'd like to tackle all that in a separate issue that may require restructuring how messages are returned etc. Especially since most of these advanced errors will require us to link the user to an external page which isn't really user-friendly at the moment and really painful.

The problem here, specifically is, the validator cannot return a custom error message because ajv doesn't allow that. I looked at that briefly and it's quite hard to customize and may even require patching ajv and working together with upstream to get custom error messages somehow into custom validation-formats.

I'll file a follow-up issue with more details if that works for you?

Reply:
Absolutely, I agree that we can't (and we shouldn't) solve it as part of this PR, especially given that this PR has already been "exciting" enough :-)